guide/interactions/slash-command-permissions.md

```diff
@@ -0,0 +1,137 @@
+
+	if (message.content.toLowerCase() === '!perms' && message.author.id === client.application?.owner.id) {
+		const command = await client.guilds.cache.get('123456789012345678')?.commands.fetch('876543210987654321');
+
+		const permissions = [
+			{
+				id: '224617799434108928',
+				type: 'USER',
+				permission: true,
+			},
+		];
+
+		await command.setPermissions({ permissions });
+	}
+});
+```
+
+And that's how you use slash command permissions to deny or allow access to a single user!
+
+## Role permissions
+
+Now you may want to allow (or deny) multiple users the usage of a command. In this scenario you can apply a permission that is scoped to a role instead of a user:
+
+```js {10,11}
+client.on('messageCreate', async message => {
+	if (!client.application?.owner) await client.application?.fetch();
+
+	if (message.content.toLowerCase() === '!perms' && message.author.id === client.application?.owner.id) {
+		const command = await client.guilds.cache.get('123456789012345678')?.commands.fetch('876543210987654321');
+
+		const permissions = [
+			{
+				id: '464464090157416448',
+				type: 'ROLE',
+				permission: false,
+			},
+		];
+
+		await command.permissions.add({ permissions });
+	}
+});
+```
+
+With this, you have successfully denied an entire role access to this command. If you want to allow a certain role to access this command, you repeat this procedure while setting `permission: true` in the `ApplicationCommandPermissionData` as shown below:
+
+```js {11}
+client.on('messageCreate', async message => {
+	if (!client.application?.owner) await client.application?.fetch();
+
+	if (message.content.toLowerCase() === '!perms' && message.author.id === client.application?.owner.id) {
+		const command = await client.guilds.cache.get('123456789012345678')?.commands.fetch('876543210987654321');
+
+		const permissions = [
+			{
+				id: '464464090157416448',
+				type: 'ROLE',
+				permission: true,
+			},
+		];
+
+		await command.permissions.add({ permissions });
+	}
+});
+```
+
+## Bulk update permissions
+
+If you have a lot of commands, you likely want to update their permissions in one go instead of one-by-one, for this approach you can use `GuildApplicationCommandManager#setPermissions` as outlined below:
+
+```js {5-22,24}
+client.on('messageCreate', async message => {
+	if (!client.application?.owner) await client.application?.fetch();
+
+	if (message.content.toLowerCase() === '!perms' && message.author.id === client.application?.owner.id) {
+		const fullPermissions = [
+			{
+				id: '123456789012345678',
+				permissions: [{
+					id: '224617799434108928',
+					type: 'USER',
+					permission: false,
+				}],
+			},
+			{
+				id: '876543210987654321',
+				permissions: [{
+					id: '464464090157416448',
+					type: 'ROLE',
+					permission: true,
+				}],
+			},
+		];
+
+		await client.guilds.cache.get('123456789012345678')?.commands.permissions.set({ fullPermissions });
+	}
+});
```

Comment:
```suggestion
If you have a lot of commands, you likely want to update their permissions in one go instead of one-by-one. For this approach, you can use `GuildApplicationCommandManager#setPermissions`:

<!-- eslint-skip -->

```js
const fullPermissions = [
	{
		id: '123456789012345678',
		permissions: [{
			id: '224617799434108928',
			type: 'USER',
			permission: false,
		}],
	},
	{
		id: '876543210987654321',
		permissions: [{
			id: '464464090157416448',
			type: 'ROLE',
			permission: true,
		}],
	},
];

await client.guilds.cache.get('123456789012345678')?.commands.permissions.set({ fullPermissions });
```